libs/ui/primitives/src/checkbox.tsx

```diff
@@ -4,23 +4,25 @@ import * as React from 'react'
 
 import { cn } from '@nmit-coursition/ui/utils'
 
-const Checkbox = React.forwardRef<
-  React.ElementRef<typeof CheckboxPrimitive.Root>,
-  React.ComponentPropsWithoutRef<typeof CheckboxPrimitive.Root>
->(({ className, ...props }, ref) => (
-  <CheckboxPrimitive.Root
-    ref={ref}
-    className={cn(
-      'peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground',
-      className,
-    )}
-    {...props}
-  >
-    <CheckboxPrimitive.Indicator className={cn('flex items-center justify-center text-current')}>
-      <Check className='h-4 w-4' />
-    </CheckboxPrimitive.Indicator>
-  </CheckboxPrimitive.Root>
-))
-Checkbox.displayName = CheckboxPrimitive.Root.displayName
+export interface RootProps extends React.ComponentPropsWithoutRef<typeof CheckboxPrimitive.Root> {}
 
-export { Checkbox }
+const Main = React.forwardRef<React.ElementRef<typeof CheckboxPrimitive.Root>, RootProps>(
```

Comment:
**suggestion:** We shouldn't mutate generated code.

Reply:
I reverted everything back to how Shadcn originally ships it, but I added an export for the primitive root component's type to allow extension within the design system.